include/fmt/chrono.h

```diff
@@ -430,6 +430,40 @@ auto write(OutputIt out, const std::tm& time, const std::locale& loc,
   return write_encoded_tm_str(out, string_view(buf.data(), buf.size()), loc);
 }
 
+#if FMT_SAFE_DURATION_CAST
+// throwing version of safe_duration_cast
+template <typename To, typename FromRep, typename FromPeriod>
+To fmt_safe_duration_cast(std::chrono::duration<FromRep, FromPeriod> from) {
+  int ec;
+  To to = safe_duration_cast::safe_duration_cast<To>(from, ec);
+  if (ec) FMT_THROW(format_error("cannot format duration"));
+  return to;
+}
+#endif
+
+template <typename Duration,
+          FMT_ENABLE_IF(std::is_integral<typename Duration::rep>::value)>
+std::time_t to_time_t(
+    std::chrono::time_point<std::chrono::system_clock, Duration> time_point) {
+#if FMT_SAFE_DURATION_CAST
+  return fmt_safe_duration_cast<std::chrono::duration<std::time_t>>(
+             time_point.time_since_epoch())
+      .count();
+#else
+  return std::chrono::duration_cast<std::chrono::duration<std::time_t>>(
+             time_point.time_since_epoch())
+      .count();
+#endif
+}
+
+template <typename Duration,
+          FMT_ENABLE_IF(std::is_floating_point<typename Duration::rep>::value)>
+std::time_t to_time_t(
+    std::chrono::time_point<std::chrono::system_clock, Duration> time_point) {
+  return std::chrono::duration_cast<std::chrono::duration<std::time_t>>(
+             time_point.time_since_epoch())
+      .count();
+}
```

Comment:
Considering that the two overloads of `to_time_t` are identical modulo `fmt_safe_duration_cast` I suggest merging them into one and moving the difference into `fmt_safe_duration_cast` which can just forward to `duration_cast` for floating point and when `FMT_SAFE_DURATION_CAST` is disabled.

Reply:
Are you sure? I feel like `fmt_safe_duration_cast` should always offer the same safety guarantees, there being "safe" in the name. If we did this, perhaps this should be renamed `fmt_duration_cast` instead.

Regardless, I'd still need two overloads of that, since I can't use `if constexpr` to select a different path for floats. Would that be OK with you?

It also leaves open the actual issue, being that the safe casts don't seem to be available for mixed float<->integer casts. I don't immediately see a reason why, other than it just being more work.

Reply:
Overflow in float to int conversions are also undefined behaviour, ideally we should be able to catch these in safe mode.

Reply:
Renaming to `fmt_duration_cast` sounds reasonable. We should handle the "safe" mode in it rather than having the callers deal with it. I understand that there will be two overloads, but the overall result should be simpler.

Reply:
Done and pushed